src/client.js

```diff
@@ -68,12 +68,15 @@ export class Saturn {
    * @returns {Promise<object>}
    */
   async fetchCIDWithRace (cidPath, opts = {}) {
-    const [cid] = (cidPath ?? '').split('/')
-    CID.parse(cid)
-
-    const jwt = await getJWT(this.opts, this.storage)
+    const options = Object.assign({}, this.opts, opts)
```

Comment:
Can't submit a multiline comment for some reason.

Line 71-76 references `options`, `opts`, and `this.opts`. This is too confusing and will lead to bugs. Once we create the `options` object, we should only reference that, and stop referencing `opts` and `this.opts`